sql/dd/impl/bootstrap/bootstrap_ctx.h

```diff
@@ -123,10 +124,16 @@ class DD_bootstrap_ctx {
     m_actual_I_S_version = actual_I_S_version;
   }
 
+  void set_actual_dd_engine(uint actual_dd_engine) {
+    m_actual_dd_engine = actual_dd_engine;
+  }
+
   uint get_actual_dd_version() const { return m_actual_dd_version; }
 
   uint get_actual_I_S_version() const { return m_actual_I_S_version; }
 
+  uint get_actual_dd_engine() const { return m_actual_dd_engine; }
```

Comment:
This should never be called if `set_actual_dd_engine` was not called, consider adding `assert(m_actual_dd_engine != DB_TYPE_UNKNOWN);`